contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,202 @@
+    function setIndividualRateLimitPerSecond(address rateLimitedAddress, uint112 newRateLimitPerSecond) external virtual override onlyGovernorOrAdmin {
+        require(newRateLimitPerSecond <= rateLimitPerSecond, "MultiRateLimited: rateLimitPerSecond too high");
+
+        _updateIndividualBufferStored(rateLimitedAddress);
+        _setIndividualRateLimitPerSecond(rateLimitedAddress, newRateLimitPerSecond);
+    }
+
+    /// @notice set the buffer cap
+    /// @param rateLimitedAddress the address whose buffer will be set
+    /// @param newBufferCap the new buffer cap for this rateLimitedAddress
+    function setIndividualBufferCap(address rateLimitedAddress, uint144 newBufferCap) external virtual override onlyGovernorOrAdmin {
+        require(newBufferCap <= bufferCap, "MultiRateLimited: new buffer cap is over global max");
+
+        _setIndividualBufferCap(rateLimitedAddress, newBufferCap);
+    }
+
+    /// @notice the amount of action used before hitting limit
+    /// @dev replenishes at rateLimitPerSecond per second up to bufferCap
+    /// @param rateLimitedAddress the address whose buffer will be returned
+    function individualBuffer(address rateLimitedAddress) public view override returns(uint144) {
+        RateLimitData memory rateLimitData = rateLimitPerAddress[rateLimitedAddress];
+
+        uint256 elapsed = block.timestamp - rateLimitData.lastBufferUsedTime;
+        return uint144(Math.min(rateLimitData.bufferStored + (rateLimitData.rateLimitPerSecond * elapsed), rateLimitData.bufferCap));
+    }
+
+    /// @notice the rate per second for each address
+    function individualRateLimitPerSecond(address limiter) external override view returns(uint256) {
+        return rateLimitPerAddress[limiter].rateLimitPerSecond;
+    }
+
+    /// @notice the last time the buffer was used by each address
+    function individualLastBufferUsedTime(address limiter) external override view returns(uint256) {
+        return rateLimitPerAddress[limiter].lastBufferUsedTime;
+    }
+
+    /// @notice the cap of the buffer that can be used at once
+    function individualBufferCap(address limiter) external override view returns(uint256) {
+        return rateLimitPerAddress[limiter].bufferCap;
+    }
+
+    /// @notice the method that enforces the rate limit. Decreases buffer by "amount". 
+    /// If buffer is <= amount either
+    /// 1. Does a partial depletion by the amount remaining in the buffer or
+    /// 2. Reverts
+    /// Depending on whether doPartialAction is true or false
+    /// @param rateLimitedAddress the address whose buffer will be depleted
+    /// @param amount the amount to remove from the rateLimitedAddress's buffer
+    function _depleteBuffer(address rateLimitedAddress, uint256 amount) internal returns(uint256) {
+        _depleteBuffer(amount);
+
+        uint256 newBuffer = individualBuffer(rateLimitedAddress);
+        
+        uint256 usedAmount = amount;
```

Comment:
I think we talked about removing partial mints on Discord, thoughts?

Reply:
I don't remember us reaching a consensus on this

Reply:
For multi minter I think we agreed yeah

Reply:
having it be a global var across all minters also limits utility

Reply:
ok, removed the partial buffer depletion from MultiRateLimited